voyager-navigator/src/commonMain/kotlin/cafe/adriel/voyager/navigator/Navigator.kt

```diff
@@ -114,33 +116,41 @@ public class Navigator @InternalVoyagerApi constructor(
     public val parent: Navigator? = null
 ) : Stack<Screen> by screens.toMutableStateStack(minSize = 1) {
 
-    public val level: Int =
-        parent?.level?.inc() ?: 0
+    public var isNavigating: Boolean by mutableStateOf(false)
+    public val level: Int = parent?.level?.inc() ?: 0
 
     public val lastItem: Screen by derivedStateOf {
         lastItemOrNull ?: error("Navigator has no screen")
     }
 
+
     private val stateKeys = ThreadSafeSet<String>()
 
     internal val children = ThreadSafeMap<NavigatorKey, Navigator>()
 
+    private val results = mutableStateMapOf<String, Any?>()
+
     @Deprecated(
-        message = "Use 'lastItem' instead. Will be removed in 1.0.0.",
-        replaceWith = ReplaceWith("lastItem")
+        message = "Use 'lastItem' instead. Will be removed in 1.0.0.", replaceWith = ReplaceWith("lastItem")
     )
     public val last: Screen by derivedStateOf {
         lastItem
     }
 
+    @OptIn(ExperimentalVoyagerApi::class, InternalVoyagerApi::class)
     @Composable
     public fun saveableState(
-        key: String,
-        screen: Screen = lastItem,
-        content: @Composable () -> Unit
+        key: String, screen: Screen = lastItem, content: @Composable () -> Unit
     ) {
         val stateKey = "${screen.key}:$key"
-        stateKeys += stateKey
+        if (!(screen.key.contains("onboarding", ignoreCase = true) && stateKey.contains(
+                "onboarding", ignoreCase = true
+            ))
+        ) {
+            stateKeys += stateKey
+        }
```

Comment:
Why this "onboarding" validation here?

Reply:
this was our own implementation we can ignore this